bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,398 @@
+            outdoorTemperature = Objects.requireNonNull(outdoorTemperature.toUnit(ImperialUnits.FAHRENHEIT));
+        }
+
+        updateChannel(CHANNEL_TARGET_TEMPERATURE, targetTemperature);
+        updateChannel(CHANNEL_ALTERNATE_TARGET_TEMPERATURE, alternateTemperature);
+        updateChannel(CHANNEL_INDOOR_TEMPERATURE, indoorTemperature);
+        updateChannel(CHANNEL_OUTDOOR_TEMPERATURE, outdoorTemperature);
+    }
+
+    @Override
+    public void discovered(DiscoveryResult discoveryResult) {
+        logger.debug("Discovered {}", thing.getUID());
+        Map<String, Object> discoveryProps = discoveryResult.getProperties();
+        Configuration configuration = editConfiguration();
+
+        Object propertyDeviceId = Objects.requireNonNull(discoveryProps.get(CONFIG_DEVICEID));
+        configuration.put(CONFIG_DEVICEID, propertyDeviceId.toString());
+
+        Object propertyIpPort = Objects.requireNonNull(discoveryProps.get(CONFIG_IP_PORT));
+        configuration.put(CONFIG_IP_PORT, propertyIpPort.toString());
+
+        Object propertyVersion = Objects.requireNonNull(discoveryProps.get(CONFIG_VERSION));
+        BigDecimal bigDecimalVersion = new BigDecimal((String) propertyVersion);
+        logger.trace("Property Version in Handler {}", bigDecimalVersion.intValue());
+        configuration.put(CONFIG_VERSION, bigDecimalVersion.intValue());
+
+        updateConfiguration(configuration);
+
+        properties = editProperties();
+
+        Object propertySN = Objects.requireNonNull(discoveryProps.get(PROPERTY_SN));
+        properties.put(PROPERTY_SN, propertySN.toString());
+
+        Object propertySSID = Objects.requireNonNull(discoveryProps.get(PROPERTY_SSID));
+        properties.put(PROPERTY_SSID, propertySSID.toString());
+
+        Object propertyType = Objects.requireNonNull(discoveryProps.get(PROPERTY_TYPE));
+        properties.put(PROPERTY_TYPE, propertyType.toString());
+
+        updateProperties(properties);
+        initialize();
+    }
+
+    /**
+     * Gets the token and key from the Cloud
+     * 
+     * @param cloudProvider Cloud Provider account
+     */
+    public void getTokenKeyCloud(CloudProviderDTO cloudProvider) {
+        CloudDTO cloud = getClouds().get(config.email, config.password, cloudProvider);
+        if (cloud != null) {
+            cloud.setHttpClient(httpClient);
+            if (cloud.login()) {
+                TokenKey tk = cloud.getToken(config.deviceId);
+                Configuration configuration = editConfiguration();
+
+                configuration.put(CONFIG_TOKEN, tk.token());
+                configuration.put(CONFIG_KEY, tk.key());
+                updateConfiguration(configuration);
+
+                logger.trace("Token: {}", tk.token());
+                logger.trace("Key: {}", tk.key());
+                logger.info("Token and Key obtained from cloud, saving, back to initialize");
```

Comment:
Prevend logging to info at all places
```suggestion
                logger.debug("Token and Key obtained from cloud, saving, back to initialize");
```

Reply:
ok